packages/world/src/systems/HitPlayerSystem.sol

```diff
@@ -66,26 +67,27 @@ contract HitPlayerSystem is System {
     // Add target's total damage to target's local pool
     addEnergyToLocalPool(targetCoord, damage);
 
-    _requireHitsAllowed(caller, target, targetCoord, toolData.tool, damage, extraData);
+    // Track damage dealt
+    PlayerProgressUtils.trackHitPlayer(caller, damage);
+
+    HitPlayerLib._requireHitsAllowed(caller, target, targetCoord, toolData.tool, damage, extraData);
 
     notify(caller, HitPlayerNotification({ targetPlayer: target, targetCoord: targetCoord, damage: damage }));
   }
+}
 
+library HitPlayerLib {
   function _requireHitsAllowed(
     EntityId caller,
     EntityId target,
     Vec3 targetCoord,
     EntityId tool,
     uint128 totalDamage,
     bytes calldata extraData
-  ) internal {
+  ) public {
     // Check if target is within a force field and call hooks
     (ProgramId program, EntityId hookTarget, EnergyData memory energyData) = ForceFieldUtils.getHookTarget(targetCoord);
 
-    if (!program.exists()) {
```

Comment:
Similar question as before, why dont we need this now?

Reply:
when calling the hook it is checked internally